### PR TITLE
DBZ-7145 Introduce a new SMT to extract table schema information for jdbc sink connector

### PR DESCRIPTION
**Pull request by harveyyue**
https://issues.redhat.com/browse/DBZ-7145

@Naros @jpechane Please help to review it. Refer the schema history change schema and add the `metadata`(not sure is an appropriate name) block to payload.
I will add a IT tests for this SMT later.

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        newValue.put(metadataField, newRecordValueMetadata.metadataValue);
+
+        return record.newRecord(
+                record.topic(),
+                record.kafkaPartition(),
+                record.keySchema(),
+                record.key(),
+                newRecordValueMetadata.schema,
+                newValue,
+                record.timestamp());
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java:101]: Headers should be added too

File: debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        beforeDelegate.close();
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        Configuration config = Configuration.from(configs);
+        smtManager = new SmtManager<>(config);
+
+        if (!config.validateAndRecord(validateConfigFields(), LOGGER::error)) {
+            throw new ConnectException("Unable to validate config.");
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java:123]: Please use `DebeziumException`

File: debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            getSourceColumnComment(field.schema()).ifPresent(s -> column.put(COMMENT_KEY, s));
+            columns.add(column);
+        });
+        table.put(COLUMNS_KEY, columns);
+        metadataStruct.put(TABLE_KEY, table);
+
+        return new NewRecordValueMetadata(newValueSchema, metadataStruct);
+    }
+
+    private SchemaBuilder copySchema(Schema source) {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java:170]: Does it make sense to extract the method to Debezium's `SchemaUtil`?

**Review by jpechane — COMMENTED** [review]
@harveyyue LGTM, thanks! I've left just few minor comments.

File: debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.schema.SchemaFactory;
+import io.debezium.schema.SchemaNameAdjuster;
+import io.debezium.util.BoundedConcurrentHashMap;
+
+public class ExtractSchemaToNewRecord<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExtractSchemaToNewRecord.class);
+    public static final String METADATA_KEY = "metadata";
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/transforms/ExtractSchemaToNewRecord.java:54]: Maybe `sourceSchema` would be better name?

**Comment by harveyyue** [comment]
@jpechane Done with these review changes.

**Comment by jpechane** [comment]
@harveyyue Applied, thanks a lot!

**Pull request merged**